crates/bevy_scene/src/lib.rs

```diff
@@ -37,3 +38,8 @@ impl Plugin for ScenePlugin {
             .add_system_to_stage(CoreStage::PreUpdate, scene_spawner);
     }
 }
+
+#[cfg(not(feature = "serde"))]
+impl Plugin for ScenePlugin {
+    fn build(&self, _: &mut App) {}
+}
```

Comment:
Does disabling serde also cause the need to disable everything else in the plugin?
If a Plugin does nothing, unless a feature is activated, then I think an empty plugin shouldn't even be created.

Instead, bevy_internal should only use the plugin, if itself enabled the serde feature:
```diff
         #[cfg(feature = "bevy_scene")]
+        #[cfg(feature = "serde")]
         group.add(bevy_scene::ScenePlugin::default());
```

Reply:
I'm a bit nervous about this: it may not *always* be true and that sort of thing is really hard to spot in a couple years.

Reply:
Either bevy_scene requires serde to function (in which case, it shouldn't be an optional dependency) or bevy_scene does not require serde to function (in which case, we should still add the plugin).